svelte-kit-scss/src/lib/helpers/issues-search-query-builder.ts

```diff
@@ -33,6 +33,17 @@ export const splitFilterParameters = (query: string): string[] => {
   }, [] as string[]);
 };
 
+export const ensureRepoParameter = (query: string, repoName: string): string => {
+  const repoParam = splitFilterParameters(query).find((x) =>
```

Comment:
I have a feeling that this part is rather overengineered here. I don't really understand what needs to be done here. Are there any tests for these? 🤔

Reply:
I kind of wanted to see where you ended up with these, but it seems to me like we really need a few things:

A type of object that represents the search filters, I think this would include the `PREDEFINED_SEARCH_QUERY_PARAMETER_GROUPS` you have defined, and also the parametrized options like `repo:`, `is:issue`, `milestone:`, `sort:` and whatever else.

Then we would have a function that can convert from a string to an object of that type, and from an object of that type to a string.

If you had those 2 things separate, and tested, then I think you could clean up a lot of this code, because it would most involve creating URLs from the state objects, and parsing the current route into one of those objects.

What do you think?

Reply:
It seems overly complex now, and spread out.

I was hoping it would make it simpler and "all in one place"